Modules/Loadable/VesselEditor/Logic/vtkSlicerVesselEditorLogic.cxx

```diff
@@ -405,4 +405,36 @@ ::SplitTubeInSpatialObject(vtkMRMLSpatialObjectsNode* spatialNode, int tubeID, d
       break;
     }
   }
-}
\ No newline at end of file
+}
+
+//---------------------------------------------------------------------------
+void vtkSlicerVesselEditorLogic
+::RenumberTubesInSpatialObject(vtkMRMLSpatialObjectsNode* spatialNode)
+{
+  if (!spatialNode)
+  {
+    return ;
+  }
+
+  TubeNetType* spatialObject = spatialNode->GetSpatialObject();
+  char childName[] = "Tube";
+  TubeNetType::ChildrenListType* tubeList =
+    spatialObject->GetChildren(spatialObject->GetMaximumDepth(), childName);
```

Comment:
This seems wrong.  It is getting all children, children's children, etc.   Their parent tube ID's will be different that this tube.  Your code is flattening the tree?